### ISSUE TITLE
[REVIEW]: Rclean: A Tool for Writing Cleaner, More Transparent Code

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1312]
**Submitting author:** @MKLau (<a href="http://orcid.org/0000-0003-3758-2406">Matthew Lau</a>)
**Repository:** <a href="https://github.com/ropensci/Rclean" target ="_blank">https://github.com/ropensci/Rclean</a>
**Version:** v1.1.8
**Editor:** @danielskatz
**Reviewer:** @danielskatz
**Archive:** <a href="https://doi.org/10.5281/zenodo.3665732" target="_blank">10.5281/zenodo.3665732</a>

## Status

[![status](http://joss.theoj.org/papers/334d80d5508056dc6e7e17c6fd3ed5a6/status.svg)](http://joss.theoj.org/papers/334d80d5508056dc6e7e17c6fd3ed5a6)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/334d80d5508056dc6e7e17c6fd3ed5a6"><img src="http://joss.theoj.org/papers/334d80d5508056dc6e7e17c6fd3ed5a6/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/334d80d5508056dc6e7e17c6fd3ed5a6/status.svg)](http://joss.theoj.org/papers/334d80d5508056dc6e7e17c6fd3ed5a6)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/provtools/rclean" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@danielskatz, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.theoj.org/about#reviewer_guidelines. Any questions/concerns please let @danielskatz know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [ ] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/provtools/rclean">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Version:** v1.1.8
- [ ] **Authorship:** Has the submitting author (@MKLau) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @cboettig, it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01312/joss.01312/10.21105.joss.01312.pdf)

**Comment by labarba** [comment]
👋  @cboettig, @benmarwick — We'll carry out the review here. Thanks for your contribution to JOSS!

**Comment by labarba** [comment]
@benmarwick [asked](https://github.com/openjournals/joss-reviews/issues/1242#issuecomment-471356805) in the Pre-review issue whether submitting to Ropensci had been considered. Comments, @MKLau?

**Comment by MKLau** [comment]
@labarba @benmarwick 
No we hadn't thought to submit to Ropensci but I am familiar with the work there and would be interested in submitting there. 

Per "Has there been any discussion about whether this pkg might be suitable for @ropensci onboarding? Perhaps @noamross or @maelle could advise if this pkg is suitable? It it passes that, then it's auto-submitted here, if I understand correctly." 

The on-boarding process sounds very useful. Also, I am familiar with Noam Ross's work and would think that he would be suitable to review. 

I'm not quite sure that I fully understand the suggestion. Should a submission to ROpenSci be done in addition to (e.g. for on-boarding) or in-lieu of submitting to JOSS?

**Comment by labarba** [comment]
We have an agreement with ROpenSci where if your package goes through their review, it gets fast-tracked to a publication in JOSS with minor editorial checks.

**Comment by MKLau** [comment]
OK, if we go that route, what are the next steps given that we've already started the review here?

**Comment by labarba** [comment]
We can just pause the review, and wait until you ping us back!

**Comment by MKLau** [comment]
Ah, ok. That seems good. I'll look at their review process and let your know.

**Comment by labarba** [comment]
@cboettig, @benmarwick — Thank you for agreeing to review this JOSS submission. The review is now _paused_, while the author investigates going the ROpenSci route. Stay tuned!

**Comment by labarba** [comment]
@MKLau 👋  — did you look into ROpenSci? What do you want to do about this submission?

**Comment by MKLau** [comment]
@labarba Yes, I'm almost done with a couple of pre-submission edits based on the ROpenSci guidelines. Shooting to submit before the end of this month.

**Comment by maelle** [comment]
@MKLau before submission to rOpenSci you can open a [pre-submission inquiry](https://github.com/ropensci/software-review/issues/new/choose) so that the editors might assess whether your package is in scope. Thank you! :smile_cat:

**Comment by MKLau** [comment]
@maelle will do, thanks!

**Comment by MKLau** [comment]
Submitted a presub inquiry see #[300](https://github.com/ropensci/software-review/issues/300).

**Comment by danielskatz** [comment]
👋 @MKLau - what's happened in this in the last month?

**Comment by MKLau** [comment]
Hi Daniel, sorry for the slow reply I've been traveling and just saw this post. Not much happened last month, but shouldn't be more than a week to get things finished for submission. There are a few more functions that need some tests and the vignette needs to be added:  https://github.com/ProvTools/Rclean/projects/4.

**Comment by MKLau** [comment]
Hi @danielskatz @labarba @benmarwick, the package review should be good to start. I made an inquiry over in pre-submission thread https://github.com/ropensci/software-review/issues/300 to see what the best way would be to proceed. One of you might have a thought as well. Should I just re-open the closed pre-submission thread and re-label it? Or, should I start a new issue?

Thanks!

@MKLau

**Comment by labarba** [comment]
I don't know how you submit to rOpenSci, but that's what you need to do next. Once your software is reviewed there, it will be fast-tracked in JOSS.

**Comment by cboettig** [comment]
@labarba @MKLau To submit to rOpenSci, simply open an issue in https://github.com/ropensci/software-review/ as described in the README there.

**Comment by danielskatz** [comment]
👋 @MKLau - is there any news here?  After 4 months, if not, I suggest we mark this as withdrawn and let you resubmit later when you are ready - I will do this in a few days if I don't hear back from you.

**Comment by maelle** [comment]
The package is under review at rOpenSci https://github.com/ropensci/software-review/issues/327 cc handling editor @annakrystalli

**Comment by danielskatz** [comment]
Ok, thanks - it makes sense to keep it here then - please update this thread when it is accepted

**Comment by danielskatz** [comment]
@maelle & @MKLau - Has the rOpenSci package now been accepted? I see that the status is 6/approved, but am not sure if there's another step after that.

**Comment by MKLau** [comment]
Hi @danielskatz ,

It has been accepted. I’m currently finishing the transfer to ROpenSci org and revisions of the manuscript.

**Comment by maelle** [comment]
Yes it has been accepted. Cc @annakrystalli

**Comment by danielskatz** [comment]
@MKLau - please let us know when the paper is complete, so we can proceed to accept this in JOSS.  We don't need to wait for the transfer to rOpenSci, but we do need the final paper.

**Comment by MKLau** [comment]
Hi @danielskatz , just finished sorting out a couple of issues with Travis and Zenodo tracking. 

The manuscript is now good to go for review, you can find it [here](https://github.com/ropensci/Rclean/joss/paper.Rmd).

**Comment by danielskatz** [comment]
@MKLau - we need the paper to be a .md file here, with a .bib file, as shown in https://joss.readthedocs.io/en/latest/submitting.html#example-paper-and-bibliography  We will then build it with ```@whedon generate pdf``` (you can do this too to check)

**Comment by danielskatz** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
PDF failed to compile for issue #1312 with the following error: 

 /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/psych.rb:377:in `parse': (tmp/1312/joss/paper.md): mapping values are not allowed in this context at line 74 column 72 (Psych::SyntaxError)
	from /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/psych.rb:377:in `parse_stream'
	from /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/psych.rb:325:in `parse'
	from /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/psych.rb:252:in `load'
	from /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/psych.rb:473:in `block in load_file'
	from /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/psych.rb:472:in `open'
	from /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/psych.rb:472:in `load_file'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-9847f98e9ec6/lib/whedon.rb:125:in `load_yaml'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-9847f98e9ec6/lib/whedon.rb:85:in `initialize'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-9847f98e9ec6/lib/whedon/processor.rb:36:in `new'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-9847f98e9ec6/lib/whedon/processor.rb:36:in `set_paper'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-9847f98e9ec6/bin/whedon:55:in `prepare'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-9847f98e9ec6/bin/whedon:116:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `<main>'

**Comment by danielskatz** [comment]
@MKLau - please fix the problems in the .md and/or .bib files, perhaps updating them based on recent changes made for the rOpenSci process

**Comment by MKLau** [comment]
@danielskatz Ah ok, I’ll take a look.

**Comment by MKLau** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01312/joss.01312/10.21105.joss.01312.pdf)

**Comment by MKLau** [comment]
@danielskatz , looks like it should be good to go now.

**Comment by danielskatz** [comment]
thanks - I'll proofread this soon.

**Comment by danielskatz** [comment]
👋 @labarba - I now notice that you are listed as the editor - if you want to handle this from this point onward, please feel free.  If not, let me know and I'll switch it to me and finish it.

**Comment by danielskatz** [comment]
@whedon assign @danielskatz as editor

**Comment by danielskatz** [comment]
👋 @cboettig, @benmarwick - as this is an rOpenSci paper, we don't need an independent JOSS review of the software, so your help will not be called on - also, sorry for what might have been too many notifications...

**Comment by danielskatz** [comment]
@whedon accept

**Comment by whedon** [comment]
No archive DOI set. Exiting...

**Comment by danielskatz** [comment]
👋 @MKLau - At this point could you:
- [ ] Make a tagged release of your software, and list the version tag of the archived version here.
- [ ] Archive the reviewed software in Zenodo or a similar service (e.g. figshare, an institutional repository)
- [ ] Check the archival deposit (e.g., in Zenodo) has the correct metadata, this includes the title (should match the paper title) and author list (make sure the list is correct and people who only made a small fix are not on it); you may also add the authors' ORCID.
- [ ] Please list the DOI of the archived version here.

I can then move forward with accepting the submission.

**Comment by danielskatz** [comment]
Also, please merge the changes in https://github.com/ropensci/Rclean/pull/202

**Comment by MKLau** [comment]
Thanks @danielskatz for the edits. 

I've reviewed and accepted your pull request and tagged as v1.1.8. 

Here's the DOI and link to Zenodo. 

10.5281/zenodo.3665732

[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.3665732.svg)](https://doi.org/10.5281/zenodo.3665732)

**Comment by danielskatz** [comment]
@whedon set v1.1.8 as version

**Comment by whedon** [comment]
OK. v1.1.8 is the version.

**Comment by danielskatz** [comment]
@whedon set 10.5281/zenodo.3665732 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.3665732" target="_blank">10.5281/zenodo.3665732</a> is the archive.

**Comment by danielskatz** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1038/533452a is OK
- 10.1145/2602649.2602651 is OK
- 10.1073/pnas.1708290115 is OK
- 10.1038/sdata.2017.114 is OK
- 10.1038/s41567-018-0342-2 is OK
- 10.1126/science.1213847 is OK
- 10.1109/MCSE.2018.042781334 is OK
- 10.1371/journal.pcbi.1004140 is OK

MISSING DOIs

- https://doi.org/10.21105/joss.00550 may be missing for title: The drake R package: a pipeline toolkit for reproducibility and high-performance computing

INVALID DOIs

- None
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1305

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1305, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by danielskatz** [comment]
There are also some changes needed in the bib file - please merge https://github.com/ropensci/Rclean/pull/203

**Comment by danielskatz** [comment]
👋 @openjournals/dev - note the second author's name has gotten parsed incorrectly in the XML - should I accept this and then ask you to fix this manually?  Or can we do something before that?

**Comment by arfon** [comment]
@danielskatz - let's accept and then I'll fix afterwards.

**Comment by MKLau** [comment]
@danielskatz @arfon 

If the multiple middle initials is a problem, he can be cited as Thomas Pasquier, as in this [paper](https://tfjmp.org/files/publications/scidata-2017.pdf).

**Comment by danielskatz** [comment]
It's not a problem, we can handle it.  However, we are waiting on
> There are also some changes needed in the bib file - please merge ropensci/Rclean#203

**Comment by MKLau** [comment]
@danielskatz Great, pull request accepted. Thanks for those fixes.

**Comment by danielskatz** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1038/533452a is OK
- 10.1145/2602649.2602651 is OK
- 10.1073/pnas.1708290115 is OK
- 10.1038/sdata.2017.114 is OK
- 10.1038/s41567-018-0342-2 is OK
- 10.1126/science.1213847 is OK
- 10.1109/MCSE.2018.042781334 is OK
- 10.1371/journal.pcbi.1004140 is OK

MISSING DOIs

- https://doi.org/10.21105/joss.00550 may be missing for title: The drake R package: a pipeline toolkit for reproducibility and high-performance computing

INVALID DOIs

- None
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1306

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1306, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by danielskatz** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1038/533452a is OK
- 10.1145/2602649.2602651 is OK
- 10.1073/pnas.1708290115 is OK
- 10.1038/sdata.2017.114 is OK
- 10.1038/s41567-018-0342-2 is OK
- 10.1126/science.1213847 is OK
- 10.1109/MCSE.2018.042781334 is OK
- 10.1371/journal.pcbi.1004140 is OK

MISSING DOIs

- https://doi.org/10.21105/joss.00550 may be missing for title: The drake R package: a pipeline toolkit for reproducibility and high-performance computing

INVALID DOIs

- None
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1307

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1307, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by danielskatz** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1229020063447031808) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1308
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.01312](https://doi.org/10.21105/joss.01312)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by danielskatz** [comment]
👋 @arfon - over to you to fix the 2nd author in the metadata (xml file), then to close this

**Comment by danielskatz** [comment]
Thanks to @labarba for starting this process!

And congratulations to @MKLau and co-authors for this publication and the accompanying rOpenSci acceptance!

**Comment by arfon** [comment]
> 👋 @arfon - over to you to fix the 2nd author in the metadata (xml file), then to close this

Done.

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.01312/status.svg)](https://doi.org/10.21105/joss.01312)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01312">
  <img src="https://joss.theoj.org/papers/10.21105/joss.01312/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.01312/status.svg
   :target: https://doi.org/10.21105/joss.01312
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.01312/status.svg)](https://doi.org/10.21105/joss.01312)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
